locallibs/fix.py

```diff
@@ -119,3 +124,17 @@ def fix_other_things(framework_path, short_version):
     future'''
     return (ensure_current_version_link(framework_path, short_version) and
             fix_script_shebangs(framework_path, short_version))
+
+
+def fix_broken_signatures(files_relocatablized):
+    """
+    Unsign the binaries and libraries that were relocatablized to avoid
+    them having corrupted signatures.
+    """
+    for file in files_relocatablized:
+        print("Unsigning %s to avoid broken signature." % file)
+        subprocess.check_call([
+            UNSIGN_TOOL,
+            "--remove-signature",
+            file,
+        ])
```

Comment:
What happens if `codesign` returns an error?

Reply:
codesign might have a chance to fail (e.g. when xcode is not set up properly locally), I decided to simplify error handling by just making unsign behavior optional and adding a flag to turn it off. Do you suggest to add error handling here and make the output friendlier? I think either way works.

Reply:
I'm just curious what actually happens. Does it fail silently? Does error output go to stderr? I think it would be bad if it fails to remove the signature but the user has no indication of that.

Reply:
One common case I've seen is that xcode path is invalid. For example:

jjj@jjj-mbp relocatable-python % codesign  --remove-signature ./Python.framework/Versions/3.8/bin/python3.8
xcrun: error: active developer path ("/Applications/Xcode_11.2.1_fb.app/Contents/Developer1") does not exist
Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
See `man xcode-select` for more details.
./Python.framework/Versions/3.8/bin/python3.8: the codesign_allocate helper tool cannot be found or used
jjj@jjj-mbp relocatable-python % echo $?
1

 If that happens, a subprocess.CalledProcessError will be raised. In general since we use subprocess.check_call(), if codesign returns non-zero retcode, the exception will be thrown.

Reply:
Error output is not printed out, I tested the code by moving away xcode path, yet the code errors at a earlier location, and there we do not have stderr output either. If you'd like I can add the exception handling and prints out the stderr buffer. 

Ensuring correct modes for files in /Users/jjj/relocatable-python/Python.framework...
Analyzing /Users/jjj/relocatable-python/Python.framework...
Traceback (most recent call last):
  File "./make_relocatable_python_framework.py", line 96, in <module>
    main()
  File "./make_relocatable_python_framework.py", line 80, in main
    files_relocatablized = relocatablize(framework_path)
  File "/Users/jjj/relocatable-python/locallibs/relocatablizer.py", line 249, in relocatablize
    framework_data = analyze(full_framework_path)
  File "/Users/jjj/relocatable-python/locallibs/relocatablizer.py", line 203, in analyze
    prefix = base_install_name(some_dir)
  File "/Users/jjj/relocatable-python/locallibs/relocatablizer.py", line 194, in base_install_name
    install_name = get_install_name(dylib_name)
  File "/Users/jjj/relocatable-python/locallibs/relocatablizer.py", line 147, in get_install_name
    output_lines = subprocess.check_output(cmd).decode("utf-8").splitlines()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 223, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['/usr/bin/otool', '-D', '/Users/jjj/relocatable-python/Python.framework/Versions/3.8/Python']' returned non-zero exit status 1